src/rust/engine/process_execution/src/remote.rs

```diff
@@ -123,7 +129,16 @@ impl CommandRunner {
                 Err(ExecutionError::NotFinished(operation_name)) => {
                   let mut operation_request =
                     bazel_protos::operations::GetOperationRequest::new();
-                  operation_request.set_name(operation_name);
+                  operation_request.set_name(operation_name.clone());
+
+                  let default = 500;
+                  let max_wait = 5000;
+                  let backoff_period = backoff_per_operation.get(&operation_name).unwrap_or(&default).clone();
+
+                  // Each time increment wait time by default amount, but capped by max wait time.
+                  backoff_per_operation.insert(operation_name.clone(), min(max_wait, backoff_period + default));
+
+                  thread::sleep(Duration::from_millis(backoff_period.clone()));
```

Comment:
Using `thread::sleep` in an async context is a no-no. This needs to be using a Future-aware timing library... probably this one: https://tokio.rs/blog/2018-03-timers/

Reply:
tokio-timer = "0.2.0" or "0.2.1" gives:
```
error: failed to select a version for `futures` (required by `tokio-timer`):
all possible versions conflict with previously selected versions of `futures`
  version 0.1.18 in use by futures v0.1.18
  possible versions to select: 0.1.21, 0.1.20
```

Does that mean we need to bump future library first?

Reply:
I suspect that if you delete the Cargo.lock file, everything will Just Work.

Maybe also try making futures specify version 0.1.21 in every Cargo.toml file recrusively in src/rust/engine